worker/statushistorypruner/manifold.go

```diff
@@ -10,45 +10,75 @@ import (
 	"gopkg.in/juju/worker.v1"
 
 	"github.com/juju/juju/api/base"
-	"github.com/juju/juju/api/statushistory"
+	"github.com/juju/juju/environs"
 	jworker "github.com/juju/juju/worker"
 	"github.com/juju/juju/worker/dependency"
 )
 
 // ManifoldConfig describes the resources and configuration on which the
 // statushistorypruner worker depends.
 type ManifoldConfig struct {
-	APICallerName  string
-	MaxHistoryTime time.Duration
-	MaxHistoryMB   uint
-	PruneInterval  time.Duration
+	APICallerName string
+	EnvironName   string
+	PruneInterval time.Duration
+	NewWorker     func(Config) (worker.Worker, error)
+	NewFacade     func(base.APICaller) Facade
 	// TODO(fwereade): 2016-03-17 lp:1558657
 	NewTimer jworker.NewTimerFunc
 }
 
 // Manifold returns a Manifold that encapsulates the statushistorypruner worker.
 func Manifold(config ManifoldConfig) dependency.Manifold {
 	return dependency.Manifold{
-		Inputs: []string{config.APICallerName},
-		Start: func(context dependency.Context) (worker.Worker, error) {
-			var apiCaller base.APICaller
-			if err := context.Get(config.APICallerName, &apiCaller); err != nil {
-				return nil, errors.Trace(err)
-			}
-
-			facade := statushistory.NewFacade(apiCaller)
-			prunerConfig := Config{
-				Facade:         facade,
-				MaxHistoryTime: config.MaxHistoryTime,
-				MaxHistoryMB:   config.MaxHistoryMB,
-				PruneInterval:  config.PruneInterval,
-				NewTimer:       config.NewTimer,
-			}
-			w, err := New(prunerConfig)
-			if err != nil {
-				return nil, errors.Trace(err)
-			}
-			return w, nil
-		},
+		Inputs: []string{config.APICallerName, config.EnvironName},
+		Start:  config.start,
 	}
 }
+
+// start is a StartFunc for a Worker manifold.
+func (config ManifoldConfig) start(context dependency.Context) (worker.Worker, error) {
+	if err := config.Validate(); err != nil {
+		return nil, errors.Trace(err)
+	}
+	var apiCaller base.APICaller
+	if err := context.Get(config.APICallerName, &apiCaller); err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var environ environs.Environ
+	if err := context.Get(config.EnvironName, &environ); err != nil {
+		return nil, errors.Trace(err)
+	}
+	cfg := environ.Config()
```

Comment:
I have a PR immediately ready to land after this one which addresses the issue. The code here will disappear.